quickwit/quickwit-config/src/node_config/serialize.rs

```diff
@@ -370,11 +368,13 @@ pub fn node_config_for_test() -> NodeConfig {
         .to_path_buf();
     let metastore_uri = default_metastore_uri(&data_dir_uri);
     let default_index_root_uri = default_index_root_uri(&data_dir_uri);
-
+    let rest_config = RestConfig {
```

Comment:
We can move `rest_listen_addr` in there and stay backward compatible by duplicating the field in `RestConfig`, and copying from the `NodeConfig` if it's empty while displaying a deprecation warning for `1.0`. We should return an error if both are set.

Reply:
excellent idea! I was not sure of how to do it